CRM/Utils/Hook.php

```diff
@@ -2473,4 +2473,17 @@ public static function alterMailingRecipients(&$mailingObject, &$criteria, $cont
     );
   }
 
+  /**
+   * ALlow Extensions to custom process IPN hook data such as sending Google Analyitcs information based on the IPN
+   * @param array $IPNData - Array of IPN Data
+   * @return mixed
+   */
+  public static function alterIPNData($IPNData) {
```

Comment:
@seamuslee001 did you missed & here or there is a reason you don't want to pass by reference?

Reply:
At [dev/financial/issues/33](https://lab.civicrm.org/dev/financial/issues/33#note_9189), @seamuslee001 writes

> We could also ensure that the original data not get modified by not passing by reference so that the hook could only do its own processing without affecting the original data

The intention needs to be clarified.

Reply:
I think it makes sense to pass by reference - a hook should be able to alter